### PR TITLE
Fix `docker-compose build`

### LINKED ISSUE #1515: Packages do not match the hashes from the requirements file

### LINKED ISSUE #1516: Docker-compose errors at fake data and up
> Traceback (most recent call last):
>   File "/usr/lib/python2.7/runpy.py", line 163, in _run_module_as_main
>     mod_name, _Error)
>   File "/usr/lib/python2.7/runpy.py", line 102, in _get_module_details
>     loader = get_loader(mod_name)
>   File "/usr/lib/python2.7/pkgutil.py", line 464, in get_loader
>     return find_loader(fullname)
>   File "/usr/lib/python2.7/pkgutil.py", line 474, in find_loader
>     for importer in iter_importers(fullname):
>   File "/usr/lib/python2.7/pkgutil.py", line 430, in iter_importers
>     __import__(pkg)
>   File "liberapay/utils/__init__.py", line 267
> SyntaxError: Non-ASCII character '\xe2' in file liberapay/utils/__init__.py on line 267, but no encoding declared; see http://python.org/dev/peps/pep-0263/ for details

> [2019-06-03 19:09:58 +0000] [11] [ERROR] Non-ASCII character '\xe2' in file /app/liberapay/main.py on line 324, but no encoding declared; see http://python.org/dev/peps/pep-0263/ for details (main.py, line 324)
> web_1  | Traceback (most recent call last):
> web_1  |   File "/usr/local/lib/python2.7/dist-packages/gunicorn/workers/base.py", line 138, in load_wsgi
> web_1  |     self.wsgi = self.app.wsgi()
> web_1  |   File "/usr/local/lib/python2.7/dist-packages/gunicorn/app/base.py", line 67, in wsgi
> web_1  |     self.callable = self.load()
> web_1  |   File "/usr/local/lib/python2.7/dist-packages/gunicorn/app/wsgiapp.py", line 52, in load
> web_1  |     return self.load_wsgiapp()
> web_1  |   File "/usr/local/lib/python2.7/dist-packages/gunicorn/app/wsgiapp.py", line 41, in load_wsgiapp
> web_1  |     return util.import_app(self.app_uri)
> web_1  |   File "/usr/local/lib/python2.7/dist-packages/gunicorn/util.py", line 350, in import_app
> web_1  |     __import__(module)
> web_1  |   File "/app/liberapay/main.py", line 324
> web_1  | SyntaxError: Non-ASCII character '\xe2' in file /app/liberapay/main.py on line 324, but no encoding declared; see http://python.org/dev/peps/pep-0263/ for details
> web_1  | [2019-06-03 19:09:58 +0000] [13] [INFO] Booting worker with pid: 13
> web_1  | [2019-06-03 19:09:58 +0000] [13] [ERROR] Non-ASCII character '\xe2' in file /app/liberapay/main.py on line 324, but no encoding declared; see http://python.org/dev/peps/pep-0263/ for details (main.py, line 324)
> web_1  | Traceback (most recent call last):
> web_1  |   File "/usr/local/lib/python2.7/dist-packages/gunicorn/workers/base.py", line 138, in load_wsgi
> web_1  |     self.wsgi = self.app.wsgi()
> web_1  |   File "/usr/local/lib/python2.7/dist-packages/gunicorn/app/base.py", line 67, in wsgi
> web_1  |     self.callable = self.load()
> web_1  |   File "/usr/local/lib/python2.7/dist-packages/gunicorn/app/wsgiapp.py", line 52, in load
> web_1  |     return self.load_wsgiapp()
> web_1  |   File "/usr/local/lib/python2.7/dist-packages/gunicorn/app/wsgiapp.py", line 41, in load_wsgiapp
> web_1  |     return util.import_app(self.app_uri)
> web_1  |   File "/usr/local/lib/python2.7/dist-packages/gunicorn/util.py", line 350, in import_app
> web_1  |     __import__(module)
> web_1  |   File "/app/liberapay/main.py", line 324
> web_1  | SyntaxError: Non-ASCII character '\xe2' in file /app/liberapay/main.py on line 324, but no encoding declared; see http://python.org/dev/peps/pep-0263/ for details

### LINKED ISSUE #1566: botocore dependency needs to be updated
Installing base requirements gives this warning:
```
requests 2.21.0 has requirement urllib3<1.25,>=1.21.1, but you'll have urllib3 1.25.1 which is incompatible.
botocore 1.12.85 has requirement urllib3<1.25,>=1.20; python_version >= "3.4", but you'll have urllib3 1.25.1 which is incompatible.
```
Looks like `botocore` is really outdated - https://github.com/boto/botocore/blob/develop/CHANGELOG.rst

I am not sending this as a pull request, because I don't get how hashed in https://github.com/liberapay/liberapay.com/blob/master/requirements_base.txt are generated.

### PR DESCRIPTION
**Pull request by techtonik**
This makes `docker-compose build` green again.

Fixes #1566.
Fixes #1516.
Fixes #1515.
Updates #1304.

`docker-compose up` still needs some work for DB migrations, which is a subject for another PR.

### CODE REVIEW

File: requirements_dev.txt [hunk 1]
```diff
@@ -25,3 +25,33 @@ pytest-profiling==1.4.0 \
...
+            --hash=sha256:38a936c0a6d98a38bcc2d03fdaaedaba9f412879461dd2ceff8d37564d6522e4
+
+        pluggy==0.6.0 \
+            --hash=sha256:d345c8fe681115900d6da8d048ba67c25df42973bda370783cd58826442dcd7c \
+            --hash=sha256:e160a7fcf25762bb60efc7e171d4497ff1d8d2d75a3d0df7a21b76821ecbf5c5 \
+            --hash=sha256:7f8ae7f5bdf75671a718d2daf0a64b7885f74510bcd98b1a0bb420eb9a9d0cff
+
+        py==1.7.0 \
+            --hash=sha256:e76826342cefe3c3d5f7e8ee4316b80d1dd8a300781612ddbc765c17ba25a6c6 \
+            --hash=sha256:bf92637198836372b520efcba9e020c330123be8ce527e535d185ed4b6f45694
```
Comment (Changaco) [inline requirements_dev.txt]: Let's not have duplicate dependencies in different requirements files, please.

File: Dockerfile [hunk 2]
```diff
@@ -1,19 +1,16 @@
...
 
-ADD ./requirements_base.txt /requirements/pip/base.txt
-RUN pip install --require-hashes -r /requirements/pip/base.txt
+COPY requirements_*.txt /tmp/
 
-ADD ./requirements_dev.txt /requirements/pip/dev.txt
-RUN pip install --require-hashes -r /requirements/pip/dev.txt
+RUN pip3 install --require-hashes -r /tmp/requirements_base.txt
+RUN pip3 install --require-hashes -r /tmp/requirements_dev.txt
+RUN pip3 install --require-hashes -r /tmp/requirements_tests.txt
```
Comment (Changaco) [inline Dockerfile]: This should be folded into a single command:

```dockerfile
RUN pip3 install --require-hashes -r /tmp/requirements_base.txt -r /tmp/requirements_dev.txt -r /tmp/requirements_tests.txt
```

File: requirements_dev.txt (hunk 1 above)
Comment (techtonik) [reply requirements_dev.txt]: Moved everything `pytest-*` into `_tests`.

File: Dockerfile (hunk 2 above)
Comment (techtonik) [reply Dockerfile]: Folded.

**Pull request merged**